instrumentation/bunny/lib/opentelemetry/instrumentation/bunny/patches/queue.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Bunny
+      module Patches
+        # The Queue module contains the instrumentation patch the Queue#pop method.
+        module Queue
+          def pop(opts = { manual_ack: false }, &block)
+            super do |delivery_info, properties, payload|
+              return unless block
```

Comment:
I might be mistaken, but I think you want:
```suggestion
            return super unless block
            super do |delivery_info, properties, payload|
```